### PR TITLE
Editorial: Add note about submit buttons to popovertarget comments

### PR DESCRIPTION
**Pull request by scottaohara**
a note to reference what is already defined in the HTML spec

### CODE REVIEW

**Review by keithamus — APPROVED** [review]
This is correct to my knowledge and looks like a useful note to add so :+1:

**Review by spectranaut — CHANGES_REQUESTED** [review]
When I open this change locally, and respec runs in the browser -- respec can't find the "form owner" definition. 

Maybe `[=button/form owner=]` is what you want? from: https://respec.org/xref/?term=form+owner

File: html-aam/index.html [hunk 1]
```diff
@@ -13453,6 +13453,10 @@ <h4 id="att-popovertarget">`popovertarget`</h4>
                   <li>The element's `popovertargetaction` attribute is the "`auto`" state and the element is a descendant of the `popover` it is associated with.</li>
                   <!-- accessibility sibling will be defined in the ARIA specification -->
                 </ul>
+                <p class="note">
+                  A button that represents a <a data-cite="html/forms.html#concept-submit-button">submit button</a> with a
+                  <a data-html="html/form-control-infrastructure.html#form-owner">form owner</a> cannot invoke a popover.
```
Comment (scottaohara) [inline html-aam/index.html:13458]:
```suggestion
                  [=button/form owner=] cannot invoke a popover.
```

**Review by keithamus — APPROVED** [review]
Links seem corrected now. This still LGTM; @spectranaut did you want to double check this?

**Pull request merged**